homeassistant/components/smartthings/const.py

```diff
@@ -0,0 +1,28 @@
+"""Constants used by the SmartThings component and platforms."""
+import re
+
+APP_OAUTH_SCOPES = [
+    'r:devices:*'
+]
+APP_NAME_PREFIX = 'homeassistant.'
+CONF_APP_ID = 'app_id'
+CONF_INSTALLED_APP_ID = 'installed_app_id'
+CONF_INSTANCE_ID = 'instance_id'
+CONF_LOCATION_ID = 'location_id'
+DATA_MANAGER = 'manager'
+DATA_BROKERS = 'brokers'
+DOMAIN = 'smartthings'
+SIGNAL_SMARTTHINGS_UPDATE = 'smartthings_update'
+SIGNAL_SMARTAPP_PREFIX = 'smartthings_smartap_'
+SETTINGS_INSTANCE_ID = "hassInstanceId"
+SMARTTHINGS_CONFIG_FILE = 'smartthings.conf'
+SUPPORTED_PLATFORMS = [
+    'switch'
+]
+SUPPORTED_CAPABILITIES = [
+    'switch'
+]
+VAL_UID = "^(?:([0-9a-fA-F]{32})|([0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]" \
+          "{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}))$"
+VAL_UID_MATCHER = re.compile(VAL_UID)
+WEBHOOK_ID = 'smartapp'
```

Comment:
I would recommend generating a unique `webhook_id` for each instance rather than using a hardcoded one if at all possible.

Reply:
Is the concern about having a known end-point or webhook id collision?  Incoming requests are signed by SmartThings with a unique private key for the app and then validated in the pysmartapp library -- so requests would be difficult to spoofed, even if you know the URL.  The end-point has to be registered within the SmartThings infrastructure and must be live whenever set.  It would add quite a bit of complexity to support a randomly generated one each start-up, for what added value?

Reply:
I was concerned about both. If we're validating the signature from SmartThings though that removes a lot of the risk that incoming requests would be spoofed. However, I think it's still possible that there could be a collision between other webhooks the user could have created. If you would prefer to stick with a static one, I would recommend something with the `DOMAIN` in it which should avoid the collision potential.

To clarify, I was not suggesting generating a new `webhook_id` at startup every time, rather generating one when the component is setup for the first time (and storing that in the config entry). This is the workflow that we use for other components, including [dialogflow](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/dialogflow/__init__.py#L88) and [ifttt](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/ifttt/__init__.py#L100).

I imagine you could do something similar to [this](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/config_entry_flow.py#L97) in your config flow to store the webhook_id as part of your config entry.

Reply:
Makes sense.  Only one end-point for the whole component is needed, as it doesn't vary by config entry.  It could be stored to the smartthings.conf file.

Reply:
I don't have an objection with storing it there instead of in a config entry, not sure if anyone else will though.

Reply:
Addressed in latest push! Thanks!